TestCreateReadycheck flakes

### Failure cluster [2df36d06d8c30d4d34e9](https://go.k8s.io/triage#2df36d06d8c30d4d34e9)

https://testgrid.k8s.io/sig-release-master-blocking#ci-kubernetes-unit
flakes several times recently

##### Error text:
```
Failed;Failed;
=== RUN   TestCreateReadycheck/ok_if_response_time_lower_than_custom_timeout
    factory_test.go:242: healthcheck() missmatch want <nil> got etcd client connection not yet established
    --- FAIL: TestCreateReadycheck/ok_if_response_time_lower_than_custom_timeout (1.00s)
;=== RUN   TestCreateReadycheck
--- FAIL: TestCreateReadycheck (13.22s)
;
```
#### Recent failures:
[2023/8/1 04:54:44 ci-kubernetes-unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1686117640575127552)
[2023/7/31 23:08:44 ci-kubernetes-unit-1-28](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit-1-28/1686030549379977216)
[2023/7/30 21:37:31 ci-kubernetes-unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1685645253404528640)
[2023/7/29 11:29:05 ci-kubernetes-unit-eks-canary](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit-eks-canary/1685129901318868992)
[2023/7/28 10:24:48 ci-kubernetes-unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1684751208952107008)


/kind flake
/sig api-machinery

--- a/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go
@@ -157,10 +157,10 @@ func newETCD3Check(c storagebackend.Config, timeout time.Duration, stopCh <-chan
 	var prober *etcd3ProberMonitor
 	clientErr := fmt.Errorf("etcd client connection not yet established")
 
-	go wait.PollUntil(time.Second, func() (bool, error) {
-		newProber, err := newETCD3ProberMonitor(c)
+	go wait.PollImmediateUntil(time.Second, func() (bool, error) {
 		lock.Lock()
 		defer lock.Unlock()
+		newProber, err := newETCD3ProberMonitor(c)
 		// Ensure that server is already not shutting down.
 		select {
 		case <-stopCh:

